src/Infrastructure/Persistence/Context/ApplicationDbContext.cs

```diff
@@ -1,75 +1,20 @@
-using System.Data;
+using Finbuckle.MultiTenant;
 using FSH.WebApi.Application.Common.Events;
 using FSH.WebApi.Application.Common.Interfaces;
 using FSH.WebApi.Domain.Catalog.Brands;
 using FSH.WebApi.Domain.Catalog.Products;
-using FSH.WebApi.Domain.Common.Contracts;
-using FSH.WebApi.Infrastructure.Multitenancy;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
 
 namespace FSH.WebApi.Infrastructure.Persistence.Context;
 
 public class ApplicationDbContext : BaseDbContext
 {
-    private readonly IEventService _eventService;
-
-    public ApplicationDbContext(DbContextOptions options, ICurrentTenant currentTenant, ICurrentUser currentUser, ISerializerService serializer, IEventService eventService)
-        : base(options, currentTenant, currentUser, serializer)
+    public ApplicationDbContext(ITenantInfo currentTenant, DbContextOptions options, ICurrentUser currentUser, ISerializerService serializer, IOptions<DatabaseSettings> dbSettings, IEventService eventService)
+        : base(currentTenant, options, currentUser, serializer, dbSettings, eventService)
     {
-        _eventService = eventService;
     }
 
-    public IDbConnection Connection => Database.GetDbConnection();
-
     public DbSet<Product> Products => Set<Product>();
     public DbSet<Brand> Brands => Set<Brand>();
-
-    public override async Task<int> SaveChangesAsync(CancellationToken cancellationToken = default)
```

Comment:
So this part is moved to BaseDbContext?

Reply:
damn I wrote a whole comment and made a mistake and it's gone again... but yes it's moved to basedbcontext, as it was already partly there. we should also take the 2 auditing methods out of basedbcontext into an entityauditingservice or something like that, but that's for another pr another day ;-)